src/TicketCost.php

```diff
@@ -45,4 +45,50 @@ class TicketCost extends CommonITILCost
     public static $items_id  = 'tickets_id';
 
     public static $rightname        = 'ticketcost';
+
+    public function post_updateItem($history = true): void
+    {
+        parent::post_updateItem($history);
+
+        $this->verifTCOItem();
+    }
+
+    public function post_addItem(): void
+    {
+        parent::post_addItem();
+
+        $this->verifTCOItem();
+    }
+
+    public function post_purgeItem(): void
+    {
+        parent::post_purgeItem();
+
+        $this->verifTCOItem();
+    }
+
+    private function verifTCOItem(): void
+    {
+        if ($this->fields['tickets_id']) {
+            $item_ticket = new Item_Ticket();
+            $item_tickets = $item_ticket->find([
+                'tickets_id' => $this->fields['tickets_id']
+            ]);
+            foreach ($item_tickets as $it) {
+                $this->updateTCOItem($it['itemtype'], $it['items_id']);
+            }
+        }
+    }
+
+    public function updateTCOItem(string $itemtype, int $items_id): void
+    {
+        $item = getItemForItemtype($itemtype);
+        if ($item) {
+            $item->getFromDB($items_id);
```

Comment:
`getFromDB()` could fail
```suggestion
        if ($item && $item->getFromDB($items_id);
```